clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/UpsertGoogleScalingPolicyAtomicOperation.groovy

```diff
@@ -0,0 +1,124 @@
+import com.netflix.spinnaker.clouddriver.google.deploy.description.UpsertGoogleScalingPolicyDescription
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleClusterProvider
+import com.netflix.spinnaker.clouddriver.google.security.GoogleNamedAccountCredentials
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import org.springframework.beans.factory.annotation.Autowired
+
+class UpsertGoogleScalingPolicyAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "UPSERT_SCALING_POLICY"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  @Autowired
+  private GoogleOperationPoller googleOperationPoller
+
+  @Autowired
+  private GoogleClusterProvider googleClusterProvider
+
+  private final UpsertGoogleScalingPolicyDescription description
+
+  UpsertGoogleScalingPolicyAtomicOperation(UpsertGoogleScalingPolicyDescription description) {
+    this.description = description
+  }
+
+  /**
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertScalingPolicy": { "serverGroupName": "my-zonal-server-group", "region": "us-central1", "credentials": "my-google-account", "autoscalingPolicy": { "coolDownPeriodSec": 60, "cpuUtilization": { "utilizationTarget": 0.9 }, "maxNumReplicas": 3, "minNumReplicas": 2} } } ]' localhost:7002/gce/ops
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "upsertScalingPolicy": { "serverGroupName": "my-regional-server-group", "region": "us-central1", "credentials": "my-google-account", "autoscalingPolicy": { "coolDownPeriodSec": 60, "cpuUtilization": { "utilizationTarget": 0.9 }, "loadBalancingUtilization": { "utilizationTarget" : 0.6 }, "maxNumReplicas": 3, "minNumReplicas": 2, "customMetricUtilizations" : [ { "metric": "myMetric", "utilizationTarget": 0.9, "utilizationTargetType" : "DELTA_PER_SECOND" } ] } } } ]' localhost:7002/gce/ops
+  */
+
+  @Override
+  Void operate(List priorOutputs) {
+    def serverGroupName = description.serverGroupName
+    task.updateStatus BASE_PHASE, "Initializing upsert of scaling policy for $serverGroupName..."
+
+    def credentials = description.credentials
+    def project = credentials.project
+    def compute = credentials.compute
+    def accountName = description.accountName
+    def region = description.region
+    def serverGroup = GCEUtil.queryServerGroup(googleClusterProvider, accountName, region, serverGroupName)
+    def isRegional = serverGroup.regional
+    def zone = serverGroup.zone
+    def location = isRegional ? region : zone
+    def autoscaler = buildAutoscaler(credentials, project, location, serverGroupName, isRegional)
+
+    task.updateStatus BASE_PHASE, "Attempting to retrieve existing autoscaler for $serverGroupName..."
+
+    def scalingPolicies
+    if (isRegional) {
+      scalingPolicies = compute.regionAutoscalers().list(project, region).execute().getItems()
```

Comment:
I would have expected this to behave similarly to `CopyLastGoogleServerGroupAtomicOperation`. In that operation, we fetch the server group from the cache by using `GoogleClusterProvider`. Then we retrieve the Autoscaler (and all other) attributes from the server group model itself. Any reason not to do the same thing here? Should be able to reuse the same `GCEUtil.queryServerGroup()` call.